macosx/CreatorWindowController.mm

```diff
@@ -33,9 +39,10 @@ @interface CreatorWindowController ()
 @property(nonatomic) IBOutlet NSView* fProgressView;
 @property(nonatomic) IBOutlet NSProgressIndicator* fProgressIndicator;
 
-@property(nonatomic, readonly) tr_metainfo_builder* fInfo;
+@property(nonatomic, readonly) std::shared_ptr<tr_metainfo_builder> fBuilder;
```

Comment:
👀

Reply:
Yeah, that's to work around a wart that's specific to the macOS client. The GTK and Qt clients do a direct aggregation with `std::optional<tr_metainfo_builder>` (the optional wrapper being initially empty, and then populated when the user picks a source file / source folder), and that works fine.

What's different in the macOS client is it seems like `CreatorWindowController` is copied by value somewhere, which in turn tries to copy the controller's handle to `tr_metainfo_builder`. This isn't a good idea while the builder is running asynchronous operations, which is why I wrapped it in a `std::shared_ptr`. That way both the source and target instance of the controller can refer to the same builder.

It's out of my range to refactor this; I'm not entirely sure how the copy is happening. You can see it for yourself if you replace `std::shared_ptr<tr_metainfo_builder>` with a `std::optional<tr_metainfo_builder>` and then also add these lines to `libtransmission/makemeta.h`:

```diff
diff --git a/libtransmission/makemeta.h b/libtransmission/makemeta.h
index b8a25a74e..d918bdde4 100644
--- a/libtransmission/makemeta.h
+++ b/libtransmission/makemeta.h
@@ -25,6 +25,11 @@ class tr_metainfo_builder
 public:
     tr_metainfo_builder(std::string_view single_file_or_parent_directory);
 
+    tr_metainfo_builder(tr_metainfo_builder&&) = delete;
+    tr_metainfo_builder(tr_metainfo_builder const&) = delete;
+    tr_metainfo_builder& operator=(tr_metainfo_builder&&) = delete;
+    tr_metainfo_builder& operator=(tr_metainfo_builder const&) = delete;
+
     /*
      * Checksums must be generated before calling benc() or save().
      * Since the process is slow -- it's equivalent to "verify local data" --
```

Reply:
update: I added those `= delete` lines in https://github.com/transmission/transmission/pull/3565/commits/92b4af6f9253f037f16620b124374ef079b70ab1